tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionStringTests.cs

```diff
@@ -514,6 +514,59 @@ public void When_a_database_name_is_specified(string connectionString, string db
             subject.DatabaseName.Should().Be(db);
         }
 
+        [Theory]
+        [InlineData("mongodb://localhost/?directConnection=true&replicaSet=yeah", true, ClusterConnectionMode.Direct)]
+        [InlineData("mongodb://localhost/?directConnection=true", true, ClusterConnectionMode.Direct)]
+        [InlineData("mongodb://localhost/?directConnection=false&replicaSet=yeah", false, ClusterConnectionMode.ReplicaSet)]
+        [InlineData("mongodb://localhost/?directConnection=false", false, ClusterConnectionMode.Automatic)]
+        public void When_a_directConenction_is_specified(string connectionString, bool directConnection, ClusterConnectionMode mode)
+        {
+            var subject = new ConnectionString(connectionString);
+
+            var result = bool.Parse(subject.GetOption("directConnection"));
+            result.Should().Be(directConnection);
+        }
+
+        [Theory]
+        [InlineData("mongodb+srv://localhost/?directConnection=false", false)]
+        [InlineData("mongodb+srv://localhost/?directConnection=true", true)]
+        public void When_a_directConnection_is_specified_with_a_srv_scheme(string connectionString, bool shouldThrow)
+        {
+            ConnectionString result = null;
+            var exception = Record.Exception(() => result = new ConnectionString(connectionString));
+
+            if (shouldThrow)
+            {
+                exception.Should().BeOfType<MongoConfigurationException>();
+            }
+            else
+            {
+                exception.Should().BeNull();
+                var directConnection = bool.Parse(result.GetOption("directConnection"));
+                directConnection.Should().BeFalse();
+            }
+        }
+
+        [Theory]
+        [InlineData("mongodb://localhost1,localhost2/?directConnection=false", false)]
+        [InlineData("mongodb://localhost1,localhost2/?directConnection=true", true)]
+        public void When_a_directConnection_is_specified_with_multiple_seeds(string connectionString, bool shouldThrow)
```

Comment:
Rename method to:

    When_a_directConnection_is_specified_with_multiple_hosts

"seeds" only applies to replica sets. Multiple hosts is more generic (e.g. it applies to shard routers also).